src/main/java/uk/gov/hmcts/ccd/domain/service/createcase/DefaultCreateCaseOperation.java

```diff
@@ -117,15 +121,19 @@ public CaseDetails createCaseDetails(final String caseTypeId,
             caseTypeDefinition.getJurisdictionDefinition(),
             caseTypeDefinition);
 
+        globalSearchProcessorService.populateGlobalSearchData(caseTypeDefinition, caseDataContent.getData());
```

Comment:
fixed in b1524eb, the globalSearchProcessorService is now called twice - once before any callbacks (this ensures that data classification and data validation occurs), it is then called again after call to submit callback - where it will recalculate and replace any stale `SearchCriteria` entries in the case data